Added cmake3 binary name to search for external

OliverPerks opened the pull request:
Support for search for 'cmake' and 'cmake3' (as we see it in yum based OSs)

Commit:
Added cmake3 binary name to search for external

adamjstewart commented on var/spack/repos/builtin/packages/cmake/package.py:
```diff
@@ -24,7 +24,7 @@ class Cmake(Package):
     tags = ['build-tools']
 
-    executables = ['^cmake$']
+    executables = ['^cmake$', '^cmake3$']
```
```suggestion
    executables = ['^cmake[0-9]*$']
```
Wouldn't this be more robust (e.g., cmake 2.8, 4.0, etc.)?

Commit:
Update var/spack/repos/builtin/packages/cmake/package.py

TBH I dont know how cmake is handling binary names going forward, but I dont see a problem in a more robust version.

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

adamjstewart commented:
Can you resolve the merge conflicts?

alalazo commented:
Ping @OliverPerks

bernhardkaindl reviewed (dismissed):
Only needed for RHEL7/CentOS7 because RHEL8 make cmake3 the default. But as RHEL7 is still maintained, lets merge it.

bernhardkaindl committed:
Merge branch 'develop' into modified/cmake

bernhardkaindl commented:
@spackbut fix style

bernhardkaindl commented:
@spackbot fix style

OliverPerks committed:
[@spackbot] updating style on behalf of OliverPerks

bernhardkaindl reviewed (approved):
Suggestion by adamsteward was integrated and conflict with develop resolved.

alalazo commented:
@spackbot run pipeline

alalazo committed:
Merge branch 'develop' into modified/cmake

alalazo commented:
Force merged, since this one cannot make a pipeline build fail that previously wasn't

Jordan474 commented:
This actually broke bootstrap on rhel7 for me: because it detects `cmake3` as available external, but then the clingo package runs `cmake`, which is cmake 2.

Many other packages could be affected by this when one does `spack external find`.

I guess one could change `Cmake.setup_dependent_package` to set the the `cmake` module function to `bin/cmake3`, but I'm not sure how robust this will be, and if this will cover all cmake usages. I would argue it is safer to ignore `cmake3` and only detect "vanilla" cmake installation `bin/cmake` ?

sebrowne commented:
> This actually broke bootstrap on rhel7 for me: because it detects `cmake3` as available external, but then the clingo package runs `cmake`, which is cmake 2.
> 
> Many other packages could be affected by this when one does `spack external find`.
> 
> I guess one could change `Cmake.setup_dependent_package` to set the the `cmake` module function to `bin/cmake3`, but I'm not sure how robust this will be, and if this will cover all cmake usages. I would argue it is safer to ignore `cmake3` and only detect "vanilla" cmake installation `bin/cmake` ?

We encountered this as well at our site.
